rasa/core/actions/forms.py

```diff
@@ -520,7 +520,44 @@ async def validate(
                 self.name(),
                 f"Failed to extract slot {slot_to_fill} with action {self.name()}",
             )
-        return validation_events
+        # Add slot_filled_action
+        slot_filled_action_events = []
+        if some_slots_were_validated:
+            # get mapping for requested slot
+            requested_slot_mappings = self.get_mappings_for_slot(
+                slot_to_fill, domain
+            )
+
+            if len(requested_slot_mappings) > 0:
+                slot_filled_action_name = None
+                for mapping in requested_slot_mappings:
+                    slot_filled_action_name = mapping.get("slot_filled_action")
+
+                    if slot_filled_action_name:
+                        break
+
+                if slot_filled_action_name:
```

Comment:
TODO: Fix issue where if an "other" slot is filled, this will call the wrong action.

Reply:
Fixed.